cli/lsp/config.rs

```diff
@@ -643,34 +653,96 @@ impl Config {
   pub fn enabled_urls(&self) -> Vec<Url> {
     let mut urls = vec![];
     for (workspace_uri, _) in &self.workspace_folders {
+      let Ok(workspace_path) = specifier_to_file_path(workspace_uri) else {
+        lsp_log!("Unable to convert uri \"{}\" to path.", workspace_uri);
+        continue;
+      };
       let specifier_settings = self.settings.specifiers.get(workspace_uri);
       let enable = specifier_settings
         .and_then(|s| s.enable)
         .or(self.settings.workspace.enable)
         .unwrap_or(self.has_config_file());
+      let disable_paths = specifier_settings
+        .map(|s| &s.disable_paths)
+        .unwrap_or(&self.settings.workspace.disable_paths);
+      let resolved_disable_paths = disable_paths
+        .iter()
+        .map(|p| workspace_path.join(p))
+        .collect::<Vec<_>>();
       let enable_paths = specifier_settings
         .and_then(|s| s.enable_paths.as_ref())
         .or(self.settings.workspace.enable_paths.as_ref());
       if let Some(enable_paths) = enable_paths {
-        let Ok(scope_path) = specifier_to_file_path(workspace_uri) else {
-          lsp_log!("Unable to convert uri \"{}\" to path.", workspace_uri);
-          return vec![];
-        };
         for path in enable_paths {
-          let path = scope_path.join(path);
+          let path = workspace_path.join(path);
           let Ok(path_uri) = specifier_from_file_path(&path) else {
             lsp_log!("Unable to convert path \"{}\" to uri.", path.display());
             continue;
           };
-          urls.push(path_uri);
+          if !resolved_disable_paths.iter().any(|p| path.starts_with(p)) {
+            urls.push(path_uri);
+          }
         }
-      } else if enable {
+      } else if enable
+        && !resolved_disable_paths
+          .iter()
+          .any(|p| workspace_path.starts_with(p))
+      {
         urls.push(workspace_uri.clone());
       }
     }
 
     // sort for determinism
     urls.sort();
+    urls.dedup();
+    urls
+  }
+
+  pub fn disabled_urls(&self) -> Vec<Url> {
```

Comment:
`Config::enabled_urls()` and `Config::disabled_urls()` are used in one place for the document finder. I think we can get rid of those and instead do things more just-in-time like the recent cleanups here. I'll do that in a future PR.